internal/backend/sessions/sessionworkflows/workflows.go

```diff
@@ -126,7 +126,21 @@ func (ws *workflows) StartWorkflow(ctx context.Context, session sdktypes.Session
 
 func (ws *workflows) sessionWorkflow(wctx workflow.Context, params *sessionWorkflowParams) error {
 	wi := workflow.GetInfo(wctx)
-	sid := params.Data.Session.ID()
+	session := params.Data.Session
+	sid := session.ID()
+
+	// get session time (it was created not long after event dispatch) and then update it with actual event time if found
+	eventTime := time.Time{}
+	if eventWrapped, ok := session.Inputs()["event"]; ok {
+		if eventUnwrapped, err := sdktypes.UnwrapValue(eventWrapped); err == nil {
+			if eventMap, ok := eventUnwrapped.(map[string]any); ok {
+				if createdAt, ok := eventMap["created_at"].(time.Time); ok {
+					eventTime = createdAt
+				}
+			}
+		}
+	}
+
 	isReplaying := workflow.IsReplaying(wctx)
 
 	l := ws.l.With(
```

Comment:
It's clearer to set logger attributes as early as possible, i.e. call `l := ws.l.With()` at the beginning of the function without `replay`, and call `l = l.With()` here with only `eventTime` and `replay`.